common/src/main/java/io/netty/util/internal/AppendableCharSequence.java

```diff
@@ -68,12 +68,12 @@ public AppendableCharSequence subSequence(int start, int end) {
 
     @Override
     public AppendableCharSequence append(char c) {
-        try {
-            chars[pos++] = c;
-        } catch (IndexOutOfBoundsException e) {
-            expand();
-            chars[pos - 1] = c;
+        if (pos == chars.length) {
+            char[] old = chars;
+            chars = new char[old.length << 1];
```

Comment:
I guess this blows up if old overflows with something like negative index not allowed?